chainer/functions/connection/deconvolution_nd.py

```diff
@@ -326,54 +326,99 @@ def deconvolution_nd(x, W, b=None, stride=1, pad=0, outsize=None,
     """N-dimensional deconvolution function.
 
     This is an implementation of N-dimensional deconvolution which generalizes
-    two-dimensional one. It takes three variables: input ``x``, the filter
-    weight ``W``, and the bias vector ``b``.
+    two-dimensional one. In most of deep learning frameworks and papers, this
+    function is called **transposed convolution**. But because of historical
+    reasons (e.g. paper by Ziller `Deconvolutional Networks`_) and backward
+    compatibility, this function is called **deconvolution** in Chainer.
+
+    .. _Deconvolutional Networks: \
+http://www.matthewzeiler.com/pubs/cvpr2010/cvpr2010.pdf
+
+    It takes three variables: the input ``x``, the filter weight ``W``, and the
+    bias vector ``b``.
+
+    Notation: here is a notation for dimensionalities.
+
+    - :math:`N` is the number of spatial dimensions.
+    - :math:`n` is the batch size.
+    - :math:`c_I` and :math:`c_O` are the number of the input and output
+      channels, respectively.
+    - :math:`d_1, d_2, ..., d_N` are the size of each axis of the input's
+      spatial dimensions, respectively.
+    - :math:`k_1, k_2, ..., k_N` are the size of each axis of the filters,
+      respectively.
+    - :math:`l_1, l_2, ..., l_N` are the size of each axis of the output's
+      spatial dimensions, respectively.
+    - :math:`p_1, p_2, ..., p_N` are the size of each axis of the spatial
+      padding size, respectively.
+
+    Let :math:`(s_1, s_2, ..., s_N)` be the stride of filter application.
+    Then, the output size :math:`(l_1, l_2, ..., l_N)` is determined by the
+    following equations:
+
+    .. math::
+
+       l_n = s_n (d_n - 1)  + k_n - 2 p_n \ \ (n = 1, ..., N)
```

Comment:
We should note that the outsize is determined by this formula only if users does not give the outsize. User-defined outsize supersedes it.